lib/basecrm/services/contacts_service.rb

```diff
@@ -103,6 +103,28 @@ def update(contact)
       Contact.new(root[:data])
     end
 
+    # Upsert a contact
+    #
+    # post '/contacts/upsert?filter_param=filter_value 
+    #
+    # Create a new contact or update an existing, based on a value of a filter or a set of filters. 
+    # At least a single filter - query parameter - is required. If no parameters are present, the request will return an error.
+    # See full docs https://developers.getbase.com/docs/rest/reference/contacts
+    #
+    # @param filters [Hash] - hash contain filters, one level deep e.g. { name: 'string', 'custom_fields[field]': 'value' }
+    # @param contact [Contact, Hash] - This object's attributes describe the object to be updated or created
+    # @return [Contact] The resulting object representing updated or created resource.
+    def upsert(filters, deal)
+      validate_upsert_filters!(filters)
+      validate_type!(deal)
+
+      attributes = sanitize(deal)
+      query_string = URI.encode_www_form(filters)
+      _, _, root = @client.post("/contacts/upsert?#{query_string}", attributes)
```

Comment:
💡 If the intention of `upsert` is to be idempotent, then you should use a `PUT`.

> The difference between PUT and POST is that PUT is idempotent: calling it once or several times successively has the same effect (that is no side effect), where successive identical POST may have additional effects, like passing an order several times.
>
> [MDN Docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST)

Reply:
That may be so, but according to the docs it Zendesk Sell's case its a POST https://developers.getbase.com/docs/rest/reference/deals